client/src/app/components/task-manager/TaskManagerDrawer.tsx

```diff
@@ -0,0 +1,256 @@
+    const { tasks } = useTaskManagerData();
+
+    const [expandedItems, setExpandedItems] = useState<number[]>([]);
+
+    const closeDrawer = () => {
+      setIsExpanded(!isExpanded);
+      setExpandedItems([]);
+    };
+
+    return (
+      <NotificationDrawer ref={ref}>
+        <NotificationDrawerHeader
+          title="Task Manager"
+          customText={`${queuedCount} queued`}
+          onClose={closeDrawer}
+        >
+          <Link to="/tasks">View All Tasks</Link>
+        </NotificationDrawerHeader>
+        <NotificationDrawerBody>
+          {tasks.length == 0 ? (
+            <EmptyState variant={EmptyStateVariant.full}>
+              <EmptyStateHeader
+                headingLevel="h2"
+                titleText="There are no queued tasks"
+                icon={<EmptyStateIcon icon={CubesIcon} />}
+              />
+              <EmptyStateBody>
+                No tasks are currently ready, postponed, blocked, pending or
+                running. Completed and cancelled tasks may be viewed on the full
+                task list.
+              </EmptyStateBody>
+              <EmptyStateFooter>
+                <EmptyStateActions>
+                  <Link to="/tasks">View All Tasks</Link>
+                </EmptyStateActions>
+              </EmptyStateFooter>
+            </EmptyState>
+          ) : (
+            <NotificationDrawerList>
+              {tasks.map((task) => (
+                <TaskItem
+                  key={task.id}
+                  task={task}
+                  expanded={expandedItems.includes(task.id)}
+                  onExpandToggle={(expand) => {
+                    setExpandedItems(
+                      expand
+                        ? [...expandedItems, task.id]
+                        : expandedItems.filter((i) => i !== task.id)
+                    );
+                  }}
+                />
+              ))}
+            </NotificationDrawerList>
+          )}
+        </NotificationDrawerBody>
+      </NotificationDrawer>
+    );
+  }
+);
+TaskManagerDrawer.displayName = "TaskManagerDrawer";
+
+const TaskStateToIcon: React.FC<{ task: TaskManagerTask }> = ({ task }) =>
+  task.state === "Ready" ? (
+    <Tooltip content="Ready">
+      <CheckCircleIcon />
+    </Tooltip>
+  ) : task.state === "Postponed" ? (
+    <Tooltip content="Postponed">
+      <PauseCircleIcon />
+    </Tooltip>
+  ) : task.state === "Pending" ? (
+    <Tooltip content="Pending">
+      <PendingIcon />
+    </Tooltip>
+  ) : task.state === "Running" ? (
+    <Tooltip content="Running">
+      <InProgressIcon />
+    </Tooltip>
+  ) : (
+    <Tooltip content="Unknown">
+      <TaskIcon />
+    </Tooltip>
+  );
+
+const TaskItem: React.FC<{
+  task: TaskManagerTask;
+  expanded: boolean;
+  onExpandToggle: (expand: boolean) => void;
+}> = ({ task, expanded, onExpandToggle }) => {
+  const starttime = dayjs(task.started ?? task.createTime);
```

Comment:
1. Using `createTime` as fallback may be confusing:  `started` seems to be set after the task actually started running. Together with `terminated` they show how long the task was executing.
2. in #1957 I've used  `dayjs(date).format() ` which is `date in ISO8601, without fraction seconds e.g. '2020-04-02T08:02:17-05:00'`.

Reply:
Humm ---- I just needed to make sure there is a date to use to display when the task is "expanded".  I'm just displaying a "XX minutes ago" kind of time with the tooltip of the actual date.  So I'm not too worried about it

Reply:
well ....if it's more "last change" then we could as well do `terminated ?? started ?? createTime`.

Reply:
Only "queued" tasks are displayed in the drawer so we only care about how long since they were created (or started)

Reply:
I opened https://github.com/konveyor/tackle2-ui/issues/1969 to capture this work once all the main task functions are merged.

Reply:
Created #1994 .